packages/backend/src/webmonetization/service.ts

```diff
@@ -0,0 +1,98 @@
+import { BaseService } from '../shared/baseService'
+import { InvoiceService } from '../invoice/service'
+import { Invoice } from '../invoice/model'
+import { WebMonetization } from './model'
+import { ok } from 'assert'
+import { DateTime } from 'luxon'
+import { AccountService } from '../account/service'
+
+export interface WebMonetizationService {
+  getCurrentInvoice(accountId: string): Promise<Invoice>
+}
+
+interface ServiceDependencies extends BaseService {
+  invoiceService: InvoiceService
+  accountService: AccountService
+}
+
+export async function createWebMonetizationService({
+  logger,
+  knex,
+  invoiceService,
+  accountService
+}: ServiceDependencies): Promise<WebMonetizationService> {
+  const log = logger.child({
+    service: 'WebMonetizationService'
+  })
+  const deps: ServiceDependencies = {
+    logger: log,
+    knex,
+    invoiceService,
+    accountService
+  }
+  return {
+    getCurrentInvoice: (id) => getCurrentInvoice(deps, id)
+  }
+}
+
+async function getCurrentInvoice(
+  deps: ServiceDependencies,
+  accountId: string
+): Promise<Invoice> {
+  const account = await deps.accountService.get(accountId)
+  if (!account) {
+    throw new Error('account not found')
+  }
+
+  const wm = await WebMonetization.query(deps.knex)
+    .findById(account.id)
+    .then((wm) => {
+      if (!wm) {
+        return WebMonetization.query(deps.knex).insertAndFetch({
+          accountId: account.id
+        })
+      }
+      return wm
+    })
+
+  ok(deps.knex)
+  const expectedExpiryAt = DateTime.utc().endOf('day') //Expire Every Day
+  // Create an invoice
+  if (!wm.currentInvoiceId) {
+    return WebMonetization.transaction(deps.knex, async (trx) => {
+      const description = 'Webmonetization earnings'
+      const invoice = await deps.invoiceService.create(
```

Comment:
This code is a little misleading. The invoice isn't actually created within the sql transaction.
(likewise for the `else` block below).

Reply:
Thanks for picking this up. I think I was going to add the invoice service taking in an optional trx but forget.

Reply:
>likewise for the else block below

I'd vote for not duplicating the code creating the new invoice if possible.

Reply:
@wilsonianb yeah agreed